### PR TITLE
Create the removeSection endpoint

### PR DESCRIPTION
**Pull request by Anurav-Garg**

### CODE REVIEW

File: src/controllers/timetable/removeSection.ts [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    if (timetable.authorId !== author.id) {
+      return res.status(403).json({ message: "user does not own timetable" });
+    }
+
+    let section: Section | null = null;
+
+    try {
+      section = await sectionRepository
+        .createQueryBuilder("section")
+        .innerJoin("section.timetables", "timetable")
```
Comment (soumitradev) [inline src/controllers/timetable/removeSection.ts:112]: do you really need this join here? Instead you can check if section id exists in timetable.sections from line 87

File: src/controllers/timetable/removeSection.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    }
+
+    let sectionTypes: SectionTypeList = [];
+
+    try {
+      const sectionTypeHolders = await sectionRepository
+        .createQueryBuilder("section")
+        .select("section.type")
+        .where("section.courseId = :courseId", { courseId: courseId })
+        .distinctOn(["section.type"])
```
Comment (soumitradev) [inline src/controllers/timetable/removeSection.ts:169]: is this distinctOn needed? don't we already assume they are distinct in the db?

**Review by soumitradev — CHANGES_REQUESTED** [review]
Other than a couple of things, LGTM

File: src/controllers/timetable/removeSection.ts (hunk 2 above)
Comment (Anurav-Garg) [reply src/controllers/timetable/removeSection.ts:169]: Yes it is needed, as this is querying all sections belonging to the course, not just the ones in the timetable

File: src/controllers/timetable/removeSection.ts (hunk 1 above)
Comment (Anurav-Garg) [reply src/controllers/timetable/removeSection.ts]: done

**Review by soumitradev — DISMISSED** [review]
LGTM, will merge after #34

**Review by soumitradev — APPROVED** [review]
Refactored some minor stuff in PR to conform to ESLint rules, looks good now.

**Pull request merged**